handlers/validator.go

```diff
@@ -308,6 +185,19 @@ func Validator(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Check if the validator has balance data in the finalized epochs
+	balances, err := db.BigtableClient.GetValidatorBalanceHistory([]uint64{index}, lastFinalizedEpoch, lastFinalizedEpoch)
```

Comment:
Right, I was worried about that.
Ok, new approach. I reverted everything and just say that it is not an error if the balance is not found.
In that case it will just remain at the initialized 0 (**CurrentBalance** & **EffectiveBalance**). This balance shouldn't be relevant before the validator is activated anyway afaik.

This does leave the possibility that something else causes an ISE after the balance calculation but I couldn't find anything else right now.